stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocket.kt

```diff
@@ -295,10 +300,14 @@ internal open class ChatSocket constructor(
         socket = null
     }
 
-    private fun callListeners(call: (SocketListener) -> Unit) {
+    private fun callListeners(forceSync: Boolean = false, call: (SocketListener) -> Unit) {
```

Comment:
Maybe we could create 2 separate methods to make this more transparent?

Kinda
```kotlin
private fun callListeners(call: (SocketListener) -> Unit) {
    synchronized(listeners) {
        listeners.forEach(call)
    }
}
```
and 
```kotlin
private fun callListenersOnUiThread(call: (SocketListener) -> Unit) {
    handler.post {
        callListeners(call)
    }
}
```